bin/rucio

```diff
@@ -1796,6 +1795,91 @@ def list_datasets_rse(args):
     return SUCCESS
 
 
+@exception_handler
+def add_lifetime_exception(args):
+    """
+    %(prog)s add_lifetime_exception [options] <field1=value1 field2=value2 ...>
+
+    Declare a list of bad replicas.
+
+    """
+    client = get_client(args)
+    if not args.reason:
+        logger.error('reason for the extension is mandatory')
+        return FAILURE
+    reason = args.reason
+    if not args.expiration:
+        logger.error('expiration is mandatory')
+        return FAILURE
+    try:
+        expiration = isoparse(args.expiration)
+    except ValueError as err:
+        logger.error(err)
+        return FAILURE
+
+    if not args.inputfile:
+        logger.error('inputfile is mandatory')
+        return FAILURE
+    with open(args.inputfile) as infile:
+        dids = list(filter(None, [line.strip() for line in infile]))
+
+    dids_list = []
+    containers = []
+    datasets = []
+    error_types = ['Total DIDs', 'DID not submitted because it is a file', 'DID not submitted because it is a container', 'DID not submitted because it is not part of the lifetime campaign', 'DID successfully submitted']
+    for did in dids:
+        scope, name = get_scope(did, client)
+        dids_list.append({'scope': scope, 'name': name})
+    summary = {0: len(dids_list), 1: 0, 2: 0, 3: 0, 4: 0}
```

Comment:
The different lists also do not seem clean. You have redundant information: `summary[2] == len(containers)', `summary[4] == len(datasets)`. The `namedtuple` seems like a nice solution, maybe add the lists there and use `len` instead of a counter.